### DISCUSSION TITLE
multiple subnet but only one establishing

### DISCUSSION THREAD

**Discussion by SnarkyPup** [strongswan/strongswan#2270] [Q&A]
System 
 - OS: [debian 12 bookworm]
 - Kernel version : [6.6.20+rpt-rpistrong-v8]
 - strongSwan version(s): [5.9.8/k6.6.20+rpt-rpi-v8]

Hello thank you for reading my inquiry 
My setup consists of a raspberry pi with two interfaces : one for the ADSL router (eth0 in 192.168.1.x) and one for the dhcp address leasing (eth1 in 10.6.x.x)  nat masquerade is enabled and traffic works through the tunnel but only one tunnel establishes if i load a second tunnel ,i've found a workaround which consists of using routes and two different left which is : %defaultroute and the ip of ETH0 192.168.1.x and the routing works only problem is that i need access to three different subnets , if i create 1 left parameter it doesnt work 

when one tunnel is up and i try to load up another one it simply fails with an error

to have access to all three subnets

ERROR:
generating CREATE_CHILD_SA, request 2 [ SA No TSi TSr ]
Parsed CREATE_CHILD_SA response 2 [ N(NO_PROP) ]
received NO_PROPOSAL_CHOSEN notify, no CHILD_SA built
Failed to establish CHILD_SA, keeping IKE_SA

The right setting is fortigate equipment and traffic is working with the workaround

Was wondering if this error was common i couldnt find any similar issue thank you 

config setup
  charondebug = "4"
  uniqueids = yes

conn %default

  type = tunnel
  auto = route
  keyexchange = ikev2
  ike = aes256-sha256-modp2048
  esp = aes256-sha256
  authby = psk
  forceencaps = yes
  rekeymargin = 3m

  

  leftsubnet = 0.0.0.0/0
  leftallowmany = yes

  right = PublicVPN-IP
  rightdns = 10.x.x.20

  keyingtries = %forever
  ikelifetime = 28800s
  lifetime = 3600s
  dpddelay = 20s
  dpdtimeout = 120s
  dpdaction = restart

conn strongswan-forti
 left = [ETH0 IP]
 rightsubnet = 10.3.x.x/24

conn net-10.4.0
 also = strongswan-forti
 left = %defaultroute
 rightsubnet=10.4.x.x/24

**Comment by tobiasbrunner** [comment]
> The right setting is fortigate equipment and traffic is working with the workaround

Then you have to configure separate connections for each combination of local and remote subnet as Fortigate does not support multiple traffic selectors per CHILD_SA.

Also, NO_PORPOSAL_CHOSEN is the wrong error if the traffic selectors are the issue, so maybe there is some other issue (or maybe it's just Fortigate doing something wrong).

**Reply by SnarkyPup** [reply]
i have tried to setup multiple tunnels but it seems to be in vain , the problem still persists i have setup 2 different tunnel and the second one does not establish i think its because of a conflict 2 vpn interface with a similar source but im not sure the second tunnel has no logs but the first one has failure logs , maybe is there a way to tag a tunnel to be chosen for the forti side ? thank you for your answer

**Comment by SnarkyPup** [comment]
okay so i added leftid to the connectors each are in the subnet of the ETH0ip and now it works but its not really practical because if we change some of the ip's it could pose a problem, its not dynamic

**Reply by tobiasbrunner** [reply]
Then chose some other identity. They don't have to be IP addresses (or at least they could be arbitrary ones and don't have to come from some specific subnet), unless that's a restriction by Fortigate.

**Reply by SnarkyPup** [reply]
okay so i chose some random ip's as a left it still works :D ! tell me if i should make a new thread , but now i have an issue im trying to get the vpn to have FQDN instead of an ip and i have the following error 
constraint check failed : identity 'foo.bar' required as i have mentionned i am using a fortigate maybe its the certificate ? 
thank you for your help ! 👍

**Reply by tobiasbrunner** [reply]
What certificate? According to your config you are using PSKs. The identities on both ends have to match (value **and** type), that's apparently not the case with whatever you configured now.

**Reply by SnarkyPup** [reply]
so it works setup the localid with the name of the fqdn of the vpn it works but now i have an issue where my peerid are not being recognized ? when i try to establish the connection 

if i have any peer id enabled on the fortigate side it works but when i have the specific peerid enabled that match my configuration it doesnt work 
and if i enable both tunnel to have any peerid it seems to chose 1 tunnels by default 

my local id on both is vpn.domain.com
